### PR TITLE
DEVEXP-402: Mount trusted-ca to registry base pki trust source

### PR DESCRIPTION
**Pull request by adambkaplan**
Ensures that no other base trust is present for the registry.

### CODE REVIEW

**Comment by adambkaplan** [comment]
/hold

Blocked by #340

**Comment by adambkaplan** [comment]
/retest

s3 issue from earlier today

File: pkg/resource/podtemplatespec.go [hunk 1]
```diff
@@ -229,7 +229,24 @@ func makePodTemplateSpec(coreClient coreset.CoreV1Interface, proxyLister configl
...
+
+	// Cluster trusted certificate authorities - mount to /usr/share/pki/ca-trust-source
+	// Registry runs update-ca-trust extract on startup, which merges the registry CAs with the cluster's trusted CAs
+	// into a single CA bundle.
+	vol = corev1.Volume{
+		Name: "trusted-ca",
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: params.TrustedCA.Name,
```
Comment (dmage) [inline pkg/resource/podtemplatespec.go]: Should it be optional? This config map doesn't exist during installation.

Comment (adambkaplan) [reply pkg/resource/podtemplatespec.go]: It will exist now that #340 merged.

**Comment by adambkaplan** [comment]
/hold

I _may_ not be mounting the trust correctly

**Comment by adambkaplan** [comment]
@dmage ready for review again

File: pkg/resource/podtemplatespec.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ import (
 
 	configapiv1 "github.com/openshift/api/config/v1"
 	configlisters "github.com/openshift/client-go/config/listers/config/v1"
-	"github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
+	v1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
```
Comment (coreydaley) [inline pkg/resource/podtemplatespec.go]: This doesn't need an alias, but it won't hurt anything

File: pkg/client/fake/fixtures.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+type Fixtures struct {
+	Listers    *client.Listers
+	KubeClient *kfake.Clientset
+}
+
+// NewFixturesBuilder initializes a new instance of FakeListersFactory
+func NewFixturesBuilder() *FixturesBuilder {
+	factory := &FixturesBuilder{
+		deploymentIndexer:          cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{}),
+		dsIndexer:                  cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{}),
```
Comment (coreydaley) [inline pkg/client/fake/fixtures.go]: I think this should be updated to daemonSetsIndexer for clarity

File: pkg/resource/podtemplatespec_test.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+	refName     string
+	mountPath   string
+	items       []corev1.KeyToPath
+	optional    bool
+}
+
+func TestMakePodTemplateSpec(t *testing.T) {
+	// TODO: Make this table-driven to verify all storage drivers
+	params := &parameters.Globals{}
+	params.Deployment.Namespace = "openshift-image-registry"
```
Comment (coreydaley) [inline pkg/resource/podtemplatespec_test.go]: v1.ImageRegistryOperatorNamespace instead of a string?

**Pull request merged**